src/Symfony/Component/Messenger/SingleHandlingTrait.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger;
+
+use Symfony\Component\Messenger\Exception\HandlerFailedException;
+use Symfony\Component\Messenger\Exception\LogicException;
+use Symfony\Component\Messenger\Stamp\HandledStamp;
+
+trait SingleHandlingTrait
+{
+    private MessageBusInterface $messageBus;
+
+    /**
+     * Dispatches the given message, expecting to be handled by a single handler
+     * and returns the result from the handler returned value.
+     * This behavior is useful for both synchronous command & query buses,
```

Comment:
fact is we're querying for a value, i'd prefer `QueryBusTrait::query`, but not a blocker :)

Reply:
I don’t want this trait’s name to imply a single use-case as there can be many (like a command bus returning IDs e.g.).